### DISCUSSION TITLE
Rejoining during the same round after going brain dead causes reapers tax + skill loss

### DISCUSSION THREAD

**Discussion by Vykn** [FakeFishGames/Barotrauma#12055] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Similar to this very old issue #3767, but if a player who disconnected becomes brain dead, then reconnects and tries to respawn during a level, they will spawn in the respawn shuttle with reapers tax and lost skills (i.e. the game thinks that they died). Their braindead body remains on the ship, so no items are lost, but it is still an issue.

This means that if a player takes longer than the `KillDisconnectedTime` to rejoin, then they must wait until the next level transition, or respawn and suffer the consequences of dying.

The rejoining player should spawn in the submarine with their items and skills intact, and the body should despawn.

### Reproduction steps

1. Start a round in multiplayer
2. Have a friend connect and then disconnect
3. Wait for them to go braindead (either wait out the 5 minutes or edit the serversettings.xml to make the `KillDisconnectedTime` very short)
4. Have the friend reconnect
5. After the respawn timer finishes (or they are manually respawned with the `respawnnow` command) they will spawn inside the respawn shuttle with none of their items and will have lost skills and gained reaper's tax

### Bug prevalence

Happens regularly

### Version

v1.0.13.2

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

This is not really a bug: the intention of the penalties is to add a cost to respawning. It doesn't matter what your cause of death was, you'll always have to choose the penalty or wait for then next round. I do agree it might make sense to allow rejoining players to respawn "for free" though, I think we could consider changing it.

**Comment by Rotiart-Drol** [comment]
This has definitely been a very big issue at least once in every campaign I've been in. Someone's internet goes down, their game crashes, anything that gets them out of the game for longer than 5 minutes, they get to eat a death penalty if they make the mistake of trying to respawn. Maybe the penalties for dying should be relaxed for victims of brain death

**Reply by Ek-Videogames** [reply]
Skill double-punishment is a feature for some reason. 🤷‍♂️ 
But this has only been the case ever since they added **keeping character skills when respawning** to the game and then removed it like 2 updates later.
https://github.com/Regalis11/Barotrauma/discussions/11204#discussion-4845978

**Reply by Regalis11** [reply]
> Skill double-punishment is a feature for some reason. 🤷‍♂️

Could you elaborate on this? What double-punishment? Doesn't seem intentional.

**Comment by Regalis11** [comment]
This has now been addressed to some extent in https://github.com/Regalis11/Barotrauma-development/commit/47a9e054bddffbd293293af17330084e84af8d25 (commit in our private repo):

> If a client joins when their character is "braindead" (killed due to disconnection), the character is revived and the client immediately regains control of it. But only if the character's vitality is above 0 - if they have received other lethal injuries or despawned, they'll have to wait for a "normal" respawn.
